### ISSUE TITLE
modprobe no longer loads facetimehd on resume

### ISSUE THREAD

**Issue by womfoo** [NixOS/nixpkgs#14885]
## Issue description

```
Apr 22 11:52:18 silverspark post-resume-start[1804]: FATAL: Could not read '/run/current-system/kernel-modules/lib/modules/4.3.6/extra/facetimehd.ko.xz': Invalid module format
Apr 22 11:52:18 silverspark systemd[1]: post-resume.service: Main process exited, code=exited, status=1/FAILURE
Apr 22 11:52:18 silverspark systemd[1]: Failed to start Post-Resume Actions.
```

### Steps to reproduce
Get a mac with facetimehd support and the following configuration.nix
```
  hardware.facetimehd.enable = true;
  nixpkgs.config.allowUnfree = true;
```
2. Close the laptop lid
3. Open the lid back (might also need to press a key)

## Technical details

I'm running unstable and assuming its broken on 16.03 too.

* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...)
16.09pre79453.32b7b00 (Flounder)

* Nix version: (run `nix-env --version`)
nix-env (Nix) 1.11.2

* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`)
"16.09pre79453.32b7b00"

**Comment by dezgeg** [comment]
It is broken since 48f51f118506 since the module is using `module-init-tools` which is obsolete and can't handle compressed modules.

**Comment by dezgeg** [comment]
Can you report if #14888 fixes this?

**Comment by womfoo** [comment]
@dezgeg yes switching to kmod works (only tested facetimehd). Are there plans porting this over to 16.03?

**Comment by vcunat** [comment]
Yeah, I assume we should port it &ndash; it doesn't seem likely to break anything and it does fix some bugs.

**Comment by grahamc** [comment]
Oh hrm. I missed this issue conversation and put in this PR: https://github.com/NixOS/nixpkgs/pull/14926 

I reckon we should backport #14888, #14883 and close #14926.

**Comment by grahamc** [comment]
This is fixed in 16.03 and master. @joachifm can you close?
